### PR TITLE
Set user params

### PR DESCRIPTION
**Pull request by runnersaw**
This does quite a bit:
- Adds server IP/URL and email fields to project creation
- Refactors cloud_api to be a singleton
- Removes a misused success flag from `get_config_without_sections` that was causing problems. (I'm not a fan of returning `(False, None)` or `(True, value)` instead of just `None` or `value`. Also, Deniz did not seem to realize this was the case (and this one even had a docstring ;) ). (I'm assuming this was Deniz because I believe these were all in the api routes. My apologies if it wasn't you))
- Suppresses the annoying and expected "ERR: get_config_with_sections" messages

Tested:
- homography
- config
- test feature
- test object
- analysis

### CODE REVIEW

File: application/app.py [hunk 1]
```diff
@@ -549,17 +544,17 @@ def saveConfig_objects(self):
...
 
     def loadConfig_objects(self):
         config_path = get_config_path()
 
-        exists, frame1 = get_config_with_sections(config_path, "config", "frame1")
-        exists, nframes = get_config_with_sections(config_path, "config", "nframes")
-        exists, mm_connection_distance = get_config_with_sections(config_path, "config", "mm-connection-distance")
-        exists, mm_segmentation_distance = get_config_with_sections(config_path, "config", "mm-segmentation-distance")
+        frame1 = get_config_with_sections(config_path, "config", "frame1")
+        nframes = get_config_with_sections(config_path, "config", "nframes")
```
Comment (dcelik) [inline application/app.py]: I think these were changed to start_frame and num_frames in dev? I will resolve the conflict as such.

**Comment by dcelik** [comment]
Interestingly enough, I went looking for the blame for  that section and found this https://github.com/santosfamilyfoundation/SantosGUI/commit/b377b68642b77a1c6bd8c7e0b97ade43bcfe1f85 see app.py diff lines 414-420. Seems you added it ;) Anyway, I think it is a splendid change, the tuple (False, None) provides little extra data in this application. I will also address some issues with your branch being behind the dev PR that fixed the API key names. Another test after the branch conflict has been resolved will be helpful.

**Review by dcelik — APPROVED** [review]

File: application/app.py [hunk 2]
```diff
@@ -549,18 +546,18 @@ def saveConfig_objects(self):
...
-        exists, frame_start = get_config_with_sections(config_path, "config", "frame_start")
-        exists, num_frames = get_config_with_sections(config_path, "config", "num_frames")
-        exists, max_connection_distance = get_config_with_sections(config_path, "config", "max_connection_distance")
-        exists, max_segmentation_distance = get_config_with_sections(config_path, "config", "max_segmentation_distance")
-
+        frame_start = get_config_with_sections(config_path, "config", "frame_start")
+        num_frames = get_config_with_sections(config_path, "config", "num_frames")
+        max_connection_distance = get_config_with_sections(config_path, "config", "max_connection_distance")
+        max_segmentation_distance = get_config_with_sections(config_path, "config", "max_segmentation_distance")
+
```
Comment (youralien) [inline application/app.py]: Delete these spaces :)

File: application/cloud_api.py [hunk 3]
```diff
@@ -53,6 +66,7 @@ def _writeIds(self):
 
     def uploadVideo(self,  video_path, identifier = None):
         print "uploadVideo called with identifier = {}".format(identifier)
+        print(self.server_addr)
```
Comment (youralien) [inline application/cloud_api.py]: leftover print statement from debugging

File: application/pm.py [hunk 4]
```diff
@@ -43,6 +42,11 @@ def __init__(self, parent):
 
         self.ui.newp_p2.registerField("video_path*", self.ui.newp_video_input)
         self.ui.newp_p2.registerField("video_start_datetime", self.ui.newp_video_start_time_input)
+        self.ui.newp_p2.registerField("video_server_input*", self.ui.newp_video_server_input)
+        self.ui.newp_p2.registerField("video_email", self.ui.newp_video_email_input)
+
+        # Set default server
+        self.ui.newp_video_server_input.setText("http://localhost:8088")
```
Comment (youralien) [inline application/pm.py:49]: this setText is redundant with the setText in the xml.

File: application/new_project.ui [hunk 5]
```diff
@@ -139,6 +139,26 @@
...
       </item>
+      <item row="2" column="0">
+       <widget class="QLabel" name="newp_video_server_label">
+        <property name="text">
+         <string>Server IP or URL</string>
+        </property>
+       </widget>
+      </item>
+      <item row="2" column="1">
+       <widget class="QLineEdit" name="newp_video_server_input" setText="http://localhost:8088"/>
```
Comment (youralien) [inline application/new_project.ui]: use one of the setText.  One of them is redundant

**Review by youralien — APPROVED** [review]
Was just pointing out comments as I saw them!  They are so minor, that its up to you if you think they are worth fixing.  Great readable code for the ip address parsing :)

**Review by jabb1123 — APPROVED** [review]

**Review by dcelik — APPROVED** [review]

**Pull request merged**